tests/operator_test.go

```diff
@@ -963,19 +887,85 @@ var _ = Describe("ALL Operator tests", func() {
 					alerts := result["data"].(map[string]interface{})["alerts"].([]interface{})
 					for _, alert := range alerts {
 						name := alert.(map[string]interface{})["labels"].(map[string]interface{})["alertname"].(string)
-						if name == "CDIOperatorDown" {
-							if state := alert.(map[string]interface{})["state"].(string); state == "firing" {
-								return true
-							}
+						if name == alertName {
+							state := alert.(map[string]interface{})["state"].(string)
+							By(fmt.Sprintf("Alert %s state %s", name, state))
+							return state == "firing"
 						}
 					}
 					return false
 				}, 10*time.Minute, 1*time.Second).Should(BeTrue())
+			}
+
+			It("[test_id:7962] CDIOperatorDown alert firing when operator scaled down", func() {
+				if !f.IsPrometheusAvailable() {
+					Skip("This test depends on prometheus infra being available")
+				}
+
+				deploymentName := "cdi-operator"
+				By("Scale down operator so alert will trigger")
+				originalReplicas := scaleDeployment(f, deploymentName, 0)
+				Eventually(func() bool {
+					dep, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					return dep.Status.Replicas == 0
+				}, 20*time.Second, 1*time.Second).Should(BeTrue())
+
+				promRule := getPrometheusRule()
+				patchPrometheusRule(promRule, "CDIOperatorDown", "1m")
+				waitForPrometheusAlert("CDIOperatorDown")
 
 				By("Ensuring original value of replicas restored")
 				scaleDeployment(f, deploymentName, originalReplicas)
-				err = utils.WaitForDeploymentReplicasReady(f.K8sClient, f.CdiInstallNs, deploymentName)
+				err := utils.WaitForDeploymentReplicasReady(f.K8sClient, f.CdiInstallNs, deploymentName)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("[test_id:9656] CDINotReady alert firing when CDI is not ready", func() {
+				if !f.IsPrometheusAvailable() {
+					Skip("This test depends on prometheus infra being available")
+				}
+
+				Eventually(func() int {
+					return getMetricValue("kubevirt_cdi_cr_ready")
+				}, 2*time.Minute, 1*time.Second).Should(BeNumerically("==", 1))
+
+				crModified = true
+				removeCDI(f, cr)
+
+				By("Creating new CDI with wrong NodeSelector")
+				cdi := &cdiv1.CDI{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cdi",
+					},
+					Spec: cr.Spec,
+				}
+				cdi.Spec.Infra.NodeSelector = map[string]string{"wrong": "wrong"}
+				cdi, err := f.CdiClient.CdiV1beta1().CDIs().Create(context.TODO(), cdi, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				promRule := getPrometheusRule()
+				patchPrometheusRule(promRule, "CDINotReady", "1m")
```

Comment:
does this work? isn't the operator overriding `1m`?

Reply:
Thanks for reminding me this one I somehow forgot. How can we prevent it? scale down thr operator?

Reply:
Following your suggestion, the updated test checks the metric instead of the alert